No retries in case of audit initialisation errors

### Issue submitter TODO list

- [x] I've looked up my issue in [FAQ](https://ui.docs.kafbat.io/faq/common-problems)
- [x] I've searched for an already existing issues [here](https://github.com/kafbat/kafka-ui/issues)
- [x] I've tried running `main`-labeled docker image and the issue still persists there
- [x] I'm running a supported version of the application which is listed [here](https://github.com/kafbat/kafka-ui/blob/main/.github/SECURITY.md)

### Describe the bug (actual behavior)

During application startup audit initialization failed. It wasn't retried and ended up producing audit event records to console (STDOUT) only, not audit topic.

### Expected behavior

Audit initialization should have some retries in place.

Also, maybe it makes sense to make this error more visible. Not sure how though. Either application could fail completely if it's not able to initialise audit feature, or print some error trace in the UI. 

### Your installation details

App version: 3074abc - v1.2.0
Helm chart version: 1.5.0
Relevant audit config:
```
kafka:
  clusters:
    - name: X
      bootstrapServers: X-1.:9093,X-2:9093,X-3:9093
      audit:
        consoleAuditEnabled: true
        topicAuditEnabled: true
        topic: sre.kafka-ui.audit-logs.log
        level: ALL
```

### Steps to reproduce

Hard to say, since I'm not quite sure what caused the error. Brokers were completely healthy during Kafka UI start.

### Screenshots

_No response_

### Logs

2025-05-21 07:34:31,673 ERROR [main] i.k.u.s.a.AuditService: -----------------------------------------------------------------
2025-05-21 07:34:31,674 ERROR [main] i.k.u.s.a.AuditService: Error initializing Audit for cluster 'X'. Audit will be disabled. See error below: 
2025-05-21 07:34:31,674 ERROR [main] i.k.u.s.a.AuditService: Error while connecting to the cluster
java.lang.IllegalStateException: Timeout on blocking read for 5000000000 NANOSECONDS
	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:129)
	at reactor.core.publisher.Mono.block(Mono.java:1807)
	at io.kafbat.ui.service.audit.AuditService.lambda$new$0(AuditService.java:61)
	at io.kafbat.ui.service.audit.AuditService.createTopicIfNeeded(AuditService.java:130)
	at io.kafbat.ui.service.audit.AuditService.createAuditWriter(AuditService.java:94)
	at io.kafbat.ui.service.audit.AuditService.<init>(AuditService.java:62)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:209)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:145)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1381)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1218)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:563)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:346)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1606)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1552)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.resolveMethodArguments(AutowiredAnnotationBeanPostProcessor.java:888)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.inject(AutowiredAnnotationBeanPostProcessor.java:849)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:146)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:509)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1445)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:600)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:346)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1155)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1121)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1056)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:66)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at io.kafbat.ui.KafkaUiApplication.startApplication(KafkaUiApplication.java:24)
	at io.kafbat.ui.KafkaUiApplication.main(KafkaUiApplication.java:17)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:102)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:64)
	at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:40)
Caused by: java.util.concurrent.TimeoutException: Timeout on blocking read for 5000000000 NANOSECONDS
	... 50 common frames omitted
2025-05-21 07:34:31,817 ERROR [main] i.k.u.s.a.AuditService: -----------------------------------------------------------------
2025-05-21 07:34:31,823 INFO  [main] i.k.u.s.a.AuditService: Audit initialization finished for cluster 'X' (console only, topic audit init failed)


### Additional context

_No response_

--- a/api/src/main/java/io/kafbat/ui/service/audit/AuditService.java
+++ b/api/src/main/java/io/kafbat/ui/service/audit/AuditService.java
@@ -31,6 +31,7 @@ import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Mono;
 import reactor.core.publisher.Signal;
+import reactor.util.retry.Retry;
 
 
 @Slf4j
@@ -38,7 +39,9 @@ import reactor.core.publisher.Signal;
 public class AuditService implements Closeable {
 
   private static final AuthenticatedUser UNKNOWN_USER = new AuthenticatedUser("Unknown", Set.of());
-  private static final Duration BLOCK_TIMEOUT = Duration.ofSeconds(5);
+  private static final Duration BLOCK_TIMEOUT = Duration.ofSeconds(35);
+  private static final Duration TOPIC_BLOCK_TIMEOUT = Duration.ofSeconds(5);
+  private static final Duration REACTIVE_TIMEOUT = Duration.ofSeconds(5);
 
   private static final String DEFAULT_AUDIT_TOPIC_NAME = "__kui-audit-log";
   private static final int DEFAULT_AUDIT_TOPIC_PARTITIONS = 1;
@@ -58,7 +61,12 @@ public class AuditService implements Closeable {
   public AuditService(AdminClientService adminClientService, ClustersStorage clustersStorage) {
     Map<String, AuditWriter> auditWriters = new HashMap<>();
     for (var cluster : clustersStorage.getKafkaClusters()) {
-      Supplier<ReactiveAdminClient> adminClientSupplier = () -> adminClientService.get(cluster).block(BLOCK_TIMEOUT);
+      Supplier<ReactiveAdminClient> adminClientSupplier = () -> adminClientService.get(cluster)
+          .timeout(REACTIVE_TIMEOUT)
+          .retryWhen(Retry.backoff(3, Duration.ofSeconds(2))
+              .maxBackoff(Duration.ofSeconds(10))
+          )
+          .block(BLOCK_TIMEOUT);
       createAuditWriter(cluster, adminClientSupplier,
           () -> MessagesService.createProducer(cluster, AUDIT_PRODUCER_CONFIG))
           .ifPresent(writer -> auditWriters.put(cluster.getName(), writer));
@@ -134,7 +142,7 @@ public class AuditService implements Closeable {
     }
     boolean topicExists;
     try {
-      topicExists = ac.listTopics(true).block(BLOCK_TIMEOUT).contains(auditTopicName);
+      topicExists = ac.listTopics(true).block(TOPIC_BLOCK_TIMEOUT).contains(auditTopicName);
     } catch (Exception e) {
       printAuditInitError(cluster, "Error checking audit topic existence", e);
       return false;
@@ -152,7 +160,7 @@ public class AuditService implements Closeable {
           .ifPresent(topicConfig::putAll);
 
       log.info("Creating audit topic '{}' for cluster '{}'", auditTopicName, cluster.getName());
-      ac.createTopic(auditTopicName, topicPartitions, null, topicConfig).block(BLOCK_TIMEOUT);
+      ac.createTopic(auditTopicName, topicPartitions, null, topicConfig).block(TOPIC_BLOCK_TIMEOUT);
       log.info("Audit topic created for cluster '{}'", cluster.getName());
       return true;
     } catch (Exception e) {

